src/test/java/io/candydoc/ddd/extract_ddd_concepts/ExtractDDDConceptsUseCaseTest.java

```diff
@@ -307,6 +311,74 @@ void find_interaction_between_two_different_concepts() throws IOException {
                 .build());
   }
 
+  @Test
+  void find_shared_kernel_inside_given_packages() throws IOException {
+    // given
+    ExtractDDDConcepts command =
+        ExtractDDDConcepts.builder()
+            .packageToScan("io.candydoc.sample.valid_bounded_contexts")
+            .build();
+
+    // when
+    extractDDDConceptsUseCase.execute(command);
+
+    // then
+    Assertions.assertThat(extractionCaptor.getResult())
+        .contains(
+            SharedKernelFound.builder()
+                .name("shared_kernel_one")
+                .description("description of shared kernel")
+                .packageName("io.candydoc.sample.valid_bounded_contexts.shared_kernel")
+                .build());
+  }
+
+  @Test
+  void find_core_concepts_inside_shared_kernel() throws IOException {
```

Comment:
ok so you have a test that cover core concepts but the implementation goes further. One rule: TDD. And the current implementation is false because it covers things that don't belong in a shared kernel.